### PR TITLE
Game struct

### PR DESCRIPTION
**Pull request by sornas**

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -53,89 +67,113 @@ fn main_loop(mut surface: GL33Surface) {
...
+         *         loop {
+         *             // ...
+         *             update(&mut self.state, delta);
+         *             // ...
+         *             draw(&mut self.state);
+         *             // ...
+         *         }
+         *     }
+         * }
+         */
```
Comment (FredTheDino) [inline src/main.rs:95]: How would this work with sylt? The bindings will require the context when they're running. We can always ignore the argument passed to the functions, but it makes our jobb harder...

File: src/main.rs [hunk 2]
```diff
@@ -20,24 +18,40 @@ pub enum Name {
...
+
+    // Load an image and a sound.
+    let coin = game.assets.load_image(Path::new("res/coin-gold.png").to_path_buf());
+    let bloop = game.assets.load_audio(Path::new("res/bloop.wav").to_path_buf());
+    //XXX game.load_image("res/coin-gold.png") and game.load_audio("res/bloop.wav")
+
+    // Add our image as a sprite sheet.
+    let coin_sheet = game.renderer.add_sprite_sheet(game.assets[coin].clone(), (16, 16));
+    //XXX I feel like load_image should do this directly. Unless we need images that aren't sprite
+    //sheets somewhere else.
```
Comment (FredTheDino) [inline src/main.rs:52]: Yeah, do this automagivally! But then the grid size needs to be passed to "load_audio".

File: src/main.rs [hunk 3]
```diff
@@ -20,24 +18,40 @@ pub enum Name {
...
+    // Create the initial game state and input manager.
+    let mut game = lingon::Game::new("game", 800, 600);
+    let mut input = bind_inputs(game.sdl());
+    //XXX The input manager is completely separate since I don't want to store it on the struct,
+    //but it still feels weird.
+
+    // Load an image and a sound.
+    let coin = game.assets.load_image(Path::new("res/coin-gold.png").to_path_buf());
+    let bloop = game.assets.load_audio(Path::new("res/bloop.wav").to_path_buf());
+    //XXX game.load_image("res/coin-gold.png") and game.load_audio("res/bloop.wav")
```
Comment (FredTheDino) [inline src/main.rs:47]: See the myriad of other comments on this. We're not solving anything, we will have to do something like this in the sylt bindings anways.

File: src/main.rs [hunk 4]
```diff
@@ -20,24 +18,40 @@ pub enum Name {
...
-    let sheet = renderer.add_sprite_sheet(assets[image].clone(), (16, 16));
+    input
+}
 
+fn main() {
+    // Create the initial game state and input manager.
+    let mut game = lingon::Game::new("game", 800, 600);
+    let mut input = bind_inputs(game.sdl());
+    //XXX The input manager is completely separate since I don't want to store it on the struct,
+    //but it still feels weird.
```
Comment (FredTheDino) [inline src/main.rs:42]: If you have a rendering context - you have already implicitly bound the input. No reason for not putting it on there.

(Assets are there aswell... They're pretty statefull and global.)

**Review by FredTheDino — APPROVED** [review]
We don't have to do the quality of life functions right now. So let's not. We still need it to be done for the sylt bindings so let's do it there.

Group the input into the same game state as the rest.

File: src/main.rs (hunk 2 above)
Comment (sornas) [reply src/main.rs]: #35

**Review by FredTheDino — APPROVED** [review]
LGTM

**Pull request merged**